File: 257.二叉树的所有路径.py
#
# @lc app=leetcode.cn id=257 lang=python
#
# [257] 二叉树的所有路径
#

# @lc code=start
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def binaryTreePaths(self, root):
        """
        :type root: TreeNode
        :rtype: List[str]
        """
        def con_tree(root, path):
            if root:
                path += str(root.val)
                if not root.left and not root.right:
                    paths.append(path)
                else:
                    path += '->'
                    con_tree(root.left, path)
                    con_tree(root.right, path)
        paths = []
        con_tree(root, '')
        return paths
# @lc code=end

File: 257.二叉树的所有路径.cpp
/*
 * @lc app=leetcode.cn id=257 lang=cpp
 *
 * [257] 二叉树的所有路径
 */

// @lc code=start
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<string> res;
    vector<string> binaryTreePaths(TreeNode* root) {
        return helper(root);
    }
    string helper(TreeNode* root){
        if(!root){
            return "";
        }
        else if(!root->left and !root->right)
        {
            
        }
        if(root->left)
        {
            return to_string(root->val) + "->" + helper(root->left);

        }
        s += to_string(root->val);
        if(root->right){
            s += to_string(root->val) + "->" + helper(root->right);
            res.push_back(s);
        }
        return s;

    }
};

// @lc code=end

